--- ./include/linux/syscalls.h
+++ ./include/linux/syscalls.h
@@ -869,6 +869,9 @@
 				      const struct iovec __user *rvec,
 				      unsigned long riovcnt,
 				      unsigned long flags);
+
+asmlinkage long sys_mcasp(int clk);
+
 asmlinkage long sys_kcmp(pid_t pid1, pid_t pid2, int type,
 			 unsigned long idx1, unsigned long idx2);
 asmlinkage long sys_finit_module(int fd, const char __user *uargs, int flags);
