--- ./sound/soc/ti/davinci-mcasp.c
+++ ./sound/soc/ti/davinci-mcasp.c
@@ -41,6 +41,8 @@
 #include "udma-pcm.h"
 #include "davinci-mcasp.h"
 
+#include <linux/syscalls.h>
+
 #define MCASP_MAX_AFIFO_DEPTH	64
 
 #ifdef CONFIG_PM
@@ -130,6 +132,8 @@
 	struct snd_pcm_hw_constraint_list chconstr[2];
 };
 
+struct snd_soc_dai *dai_global;
+
 static inline void mcasp_set_bits(struct davinci_mcasp *mcasp, u32 offset,
 				  u32 val)
 {
@@ -605,7 +609,9 @@
 
 	mcasp->dai_fmt = fmt;
 out:
-	pm_runtime_put(mcasp->dev);
+	
+
+pm_runtime_put(mcasp->dev);
 	return ret;
 }
 
@@ -707,6 +713,20 @@
 	return 0;
 }
 
+SYSCALL_DEFINE1(mcasp, int, clk)
+{
+        printk("Change MCASP clock...\n");
+	if(clk == 1)
+	{
+		davinci_mcasp_set_sysclk(dai_global, 0, 24576000, 0);
+        }
+	else
+	{
+		davinci_mcasp_set_sysclk(dai_global, 0, 11289600, 0);
+	}
+	return 0;
+}
+
 /* All serializers must have equal number of channels */
 static int davinci_mcasp_ch_constraint(struct davinci_mcasp *mcasp, int stream,
 				       int serializers)
@@ -1700,7 +1720,7 @@
 		snd_soc_add_dai_controls(dai, davinci_mcasp_iec958_ctls,
 					 ARRAY_SIZE(davinci_mcasp_iec958_ctls));
 	}
-
+	dai_global = dai;
 	return 0;
 }
 
